### DISCUSSION TITLE
checkbox binding work if value is 0 or 1 but not work when value is "0" or "1" which is normally the case for mysql && php code in backend

### DISCUSSION THREAD

**Discussion by VikramMali** [alpinejs/alpine#2452] [7. Bugs (archived)]
while in databinding of boolean property in json object to checkbox if my json object gives value for boolean object "1" insted of 1 or true , it just show checkbox unchacked, but it should be checked. Checkbox binding work file if value is 0 or 1 but not work when value is "0" or "1" which is normally the case for mysql && php code in backend

**Comment by VikramMali** [comment]
i resoved it by following code , but hope you can improve the  code at line number 2103 in cdn version:  

> `if (el.type === "checkbox") {
      if (Number.isInteger(value)) {
        el.value = value;
      } else if (!Number.isInteger(value) && !Array.isArray(value) && typeof value !== "boolean" && ![null,void 0,"1"].includes(value)) {
        el.value = String(value);
      } else {
        if (Array.isArray(value)) {
          el.checked = value.some((val) => checkedAttrLooseCompare(val, el.value));
        } else{
          el.checked = !!value;
        }
      }
    }`

**Reply by SimoTod** [reply]
I believe you have to post a small example highlighting what happens and what you are expecting to see instead, otherwise it's not too clear what you are asking for. Are you expecting the checkbox to be checked? or the value to be set to 0/1? or both?

**Reply by VikramMali** [reply]
Hi, first of all thank you for response, and thankyou for making alpinejs it is realy nice and short. Now about issue , while in databinding of boolean property in json object to checkbox if my json object gives value for boolean object "1" insted of 1 or true , it just show checkbox unchacked, but it should be checked.

**Reply by SimoTod** [reply]
I haven't created Alpine, I'm just part of the community. :)
However, yeah boolean attribute requires a boolean but it should be the other way around, it will show the checkbox checked for any string (unless I misunderstood what you are asking which is the reason why I asked for the code)
However, If you are not sure about the type and you can't convert it server-side, you can use a loose comparison with the number 0 (don't use false because `'0' != false` would not work). See https://codepen.io/SimoTod/pen/LYzNRxG

**Reply by VikramMali** [reply]
Thank you for great example, yes you understude issue right  , yes i was having idea of using loose comparison , just only problem was it need to be two way binding so i can update server database by it which is not posible when we do "loose comparison".

**Reply by SimoTod** [reply]
Ok, so you want an x-model type of thing.
In that case, you can combine x-bind and x-on to replicate it.
Something like
`<input type="checkbox" :checked="foo != 0" @input="foo = $el.checked ? '1' : '0'">`
This only binds the value on the checkbox once though. If you change `foo` later, it won't update the checkbox but it might be enough for your use case.

If you need a real 2 way binding, you can use x-effect instead:
`<input type="checkbox" x-effect="$el.checked = foo != 0" @input="foo = $el.checked ? '1' : '0'">`

Final note, if you are not bothered about keeping those value as strings, you can simply do `@input="foo = $el.checked"`

**Reply by VikramMali** [reply]
thak you , great explanation
